assets/spark-patch/Celeborn-Optimize-Skew-Partitions-spark3_2.patch

```diff
@@ -146,15 +146,59 @@ index b950c07f3d8..2cb430c3c3d 100644
  import org.apache.spark.executor.{ExecutorMetrics, TaskMetrics}
  import org.apache.spark.internal.Logging
  import org.apache.spark.internal.config
-@@ -1780,7 +1781,7 @@ private[spark] class DAGScheduler(
-           failedStage.failedAttemptIds.add(task.stageAttemptId)
-           val shouldAbortStage =
-             failedStage.failedAttemptIds.size >= maxConsecutiveStageAttempts ||
--            disallowStageRetryForTest
-+            disallowStageRetryForTest || CelebornShuffleState.isCelebornSkewedShuffle(shuffleId)
+@@ -1369,7 +1370,10 @@ private[spark] class DAGScheduler(
+     // The operation here can make sure for the partially completed intermediate stage,
+     // `findMissingPartitions()` returns all partitions every time.
+     stage match {
+-      case sms: ShuffleMapStage if stage.isIndeterminate && !sms.isAvailable =>
++      case sms: ShuffleMapStage if (stage.isIndeterminate && !sms.isAvailable) ||
```

Comment:
I think it should be `(stage.isIndeterminate || CelebornShuffleState.isCelebornSkewedShuffle(sms.shuffleDep.shuffleId)) &&  !sms.isAvailable`

Reply:
yes, done